modules/govuk/files/usr/local/bin/govuk_supervised_initctl

```diff
@@ -0,0 +1,137 @@
+#!/bin/bash
+
+# This script is a wrapper around upstart initctl start/restart/reload
+# commands and is intended for applications/service that have their main
+# process in a pid file at /var/run/$sevice/app.pid - which is the case for
+# most GOV.UK applications
+#
+# The purpose of this script is to monitor whether the running of these
+# commands results in the process actually changing - whether it be started,
+# restarted or reloaded - as this is not something running initctl has
+# sufficient visibility for (for example you can run `initctl start $app`
+# and find upstart is satisfied the app started when it crashed on boot)
+#
+# This script is pretty heavily nuanced towards GOV.UK's use cases and isn't
+# intended to be a generic initctl wrapper for all use cases. I'm also
+# somewhat convinced there are better utilities to manage this problem in
+# newer ubuntu releases.
+
+if [ "$#" -ne "2" ]; then
+  echo "Usage: sudo govuk_supervised_initctl <COMMAND> <SERVICE>"
+  exit 1
+fi
+
+set -eu
+
+status () {
+  echo "---> ${@}"
+}
+
+error () {
+  echo "ERROR: ${@}" >&2
+  exit 1
+}
+
+pid_running () {
+  ps -p ${@} &> /dev/null
+}
+
+command=$1
+service=$2
+pid_file=/var/run/$service/app.pid
+original_pid="" && [ -e $pid_file ] && original_pid=$(cat $pid_file)
+
+case $command in
+  start)
+    status "Checking if ${service} is running"
+    if [[ $(initctl status $service) =~ "running" ]]; then
```

Comment:
It's part of this condition: https://github.com/alphagov/govuk-puppet/pull/10952/files#r577065140

Reply:
Minor: although it's written in the PR comments, I think it would be helpful for someone in future to explicitly state the scenario, as it's so unintuitive e.g.

```
# Checking `initctl status` doesn't necessarily mean a GOV.UK service is
# running. For example, unicornherder processes have two PIDs: one for 
# the parent "herder" process and one for the child "master" process; the 
# former is tracked by Upstart, the latter by the pidfile. If Upstart believes 
# the process is running it will need to be stopped before another attempt 
# to start the service. This is a convenience to avoid manual intervention.
```